sign.go

```diff
@@ -23,11 +25,12 @@ const (
 func RawTxInSignature(tx *wire.MsgTx, idx int, subScript []byte,
 	hashType txscript.SigHashType, key *btcec.PrivateKey, amt int64) ([]byte, error) {
 
-	hash := calcBip143SignatureHash(subScript, txscript.NewTxSigHashes(tx), hashType, tx, idx, amt)
-	signature, err := key.Sign(hash)
-	if err != nil {
-		return nil, fmt.Errorf("cannot sign tx input: %s", err)
-	}
+	signerMap := make(map[wire.OutPoint]*wire.TxOut)
+	signerMap[tx.TxIn[idx].PreviousOutPoint] = &wire.TxOut{}
+
+	prevOuts := txscript.NewMultiPrevOutFetcher(signerMap)
```

Comment:
I think you can do: `prevOutputFetcher := txscript.NewCannedPrevOutputFetcher(subScript, amt)`.